.github/workflows/docs.yaml

```diff
@@ -0,0 +1,58 @@
+name: Build Doc
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  test_linux:
+    name: Linux
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: recursive
+
+    - name: Configuring Test Environment
+      run: |
+        sudo apt-get update
+        sudo apt-get -y install python3.9 build-essential doxygen
+        curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
+        python3.9 get-pip.py
+        python3.9 -m pip install -U pip wheel
+        python3.9 -m pip install cmake ninja
+
+        # create the _static and _templates directories under the docs/source
+        # directory if they don't exist already. This needs to be done for
+        # Sphinx to run without errors. It tries to find these directories and
+        # fails if they don't exist.
```

Comment:
These are configurable options controlled via `docs/source/conf.py`. You probably just need to configure this slightly differently, and then you won't have to create empty directories for `_static` and `_templates` to make this run without errors.

Reply:
Yes, I have changed the conf file according to the suggestions here. Let me know if this looks good to you now!